nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/ChartProcessor.java

```diff
@@ -982,9 +1002,16 @@ boolean scriptHasDataset(String script) {
 
                 String fileString = file.getAbsoluteFile().toString();
                 if (!fileString.endsWith(".nv") && !fileString.endsWith(".ucsf")) {
-                    fileString += getExtension();
+                    if (getDatasetType().equals("SPINit")) {
```

Comment:
Wouldn't it be better to use enums instead of strings for dataset types?

Reply:
Added DatasetType enum with methods for checking dataset type and adding extensions etc..  Used this to cleanup some of the code here.